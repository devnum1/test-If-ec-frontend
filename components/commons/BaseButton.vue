<template>
  <button
    type="submit"
    class="submit-button"
    :class="{
      [className]: !!className,
      disabled: disabled,
    }"
    :style="{
      width,
      height,
    }"
    :disabled="disabled"
    @click="$emit('click')"
  >
    <slot />
  </button>
</template>

<script>
export default {
  props: {
    className: {
      type: String,
      default: '',
    },
    width: {
      type: String,
      default: '',
    },
    height: {
      type: String,
      default: '',
    },
    disabled: {
      type: Boolean,
      default: false,
    },
  },
};
</script>

<style lang="scss" scoped>
.submit-button {
  color: #fff;
  display: flex;
  padding: 17px 20px;
  justify-content: center;
  align-items: center;
  border-radius: 47px;
  border: 1px solid #4b9590;
  background: #5fbbb5;
  box-shadow: 0 4px 0 0 #4b9590;
}
.disabled {
  background-color: $white;
  box-shadow: none;
  color: rgba($black, 0.15);
  border-color: rgba($black, 0.15);
}
</style>
