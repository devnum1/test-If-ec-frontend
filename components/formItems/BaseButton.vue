// 01EZTX9M3AWK0ZB16B6KYR56F1

<template>
  <button
    type="button"
    :class="[
      'btn btn--default',
      { 'btn--icon-right': iconRight },
      { 'btn--icon-plus': btnIconPlus },
      { 'btn--icon-pdf': btnIconPdf },
      { 'btn--icon-plus-label': btnIconPlusLabel },
      { 'btn--black': btnBlack },
      { 'btn--orange': btnOrange },
      { 'btn--green-outline': btnGreenOutline },
      { 'btn--black-outline': btnBlackOutline },
      btnClass,
      btnSize,
      btnFull,
    ]"
    :disabled="disabled"
    @click="callback($event)"
  >
    <i
      v-if="btnIconPlus || btnIconPlusLabel"
      class="icon-fonts icon-fonts--plus"
    />
    <i v-if="btnIconEmail" class="icon-fonts icon-fonts--mail mr-1 txt-green" />
    <span v-if="!btnIconPlus" class="btn__text" v-html="label" />
    <i v-if="iconRight" class="icon-fonts icon-fonts--arrow-right" />
    <img v-if="btnIconPdf" src="~/assets/images/pdf.svg" />
  </button>
</template>
<script>
export default {
  name: 'BaseButton',
  props: {
    label: {
      type: String,
      default: 'デフォルトボタン',
    },
    disabled: {
      type: Boolean,
      default: false,
    },
    btnBlack: {
      type: Boolean,
      default: false,
    },
    btnOrange: {
      type: Boolean,
      default: false,
    },
    btnGreenOutline: {
      type: Boolean,
      default: false,
    },
    btnBlackOutline: {
      type: Boolean,
      default: false,
    },
    btnClass: {
      type: String,
      default: '',
    },
    iconRight: {
      type: Boolean,
      default: false,
    },
    btnIconPlus: {
      type: Boolean,
      default: false,
    },
    btnIconPlusLabel: {
      type: Boolean,
      default: false,
    },
    btnSize: {
      type: String,
      default: '',
    },
    btnFull: {
      type: String,
      default: '',
    },
    btnIconEmail: {
      type: Boolean,
      default: false,
    },
    btnIconPdf: {
      type: Boolean,
      default: false,
    },
  },
  methods: {
    callback: function (e) {
      this.$emit('click', e);
    },
  },
};
</script>
<style lang="scss" scoped>
@import '~/assets/scss/components/base-buttons';
</style>
