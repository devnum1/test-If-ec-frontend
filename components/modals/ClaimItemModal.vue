// 01EZTX9M3AWK0ZB16B6KYR56F1

<template>
  <div class="claim-item-modal">
    <div class="modal__overlay">
      <div class="modal__container">
        <div class="modal__body modal-claim">
          <div class="claim-title">
            <div class="claim-title__line">お手元にご用意いただくもの</div>
          </div>
          <div class="claim-row">
            <div class="claim-row__number">
              <div class="claim-row__number__round">1</div>
            </div>
            <div class="claim-row__text">診療明細書の写真</div>
            <div class="claim-row__icon">
              <img
                class="claim-row__icon-image"
                src="~/assets/images/icon-contract.svg"
              />
            </div>
          </div>
          <div class="claim-row">
            <div class="claim-row__number">
              <div class="claim-row__number__round">2</div>
            </div>
            <div class="claim-row__text">飼い主とペットが 一緒に写った写真</div>
            <div class="claim-row__icon">
              <img
                class="claim-row__icon-image"
                src="~/assets/images/icon-pet-owner.svg"
              />
            </div>
          </div>
          <BaseButton
            btn-class="test"
            label="保険金を請求する"
            btn-full="btn--full"
            icon-right
          />
        </div>
      </div>
    </div>
  </div>
</template>
<script type="text/javascript">
export default {
  data() {
    return {
      display: true,
    };
  },
  methods: {},
};
</script>

<style lang="scss" scoped>
@import '~/assets/scss/components/modal';
</style>
