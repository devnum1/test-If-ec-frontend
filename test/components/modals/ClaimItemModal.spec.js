import { shallowMount, config } from '@vue/test-utils';
import ClaimItemModal from '@/components/modals/ClaimItemModal.vue';
import BaseButton from '@/components/formItems/BaseButton.vue';

config.mocks.$t = (key) => key;

describe('ClaimItemModal', () => {
  let wrapper;

  const factory = (computed = {}) => {
    return shallowMount(ClaimItemModal, {
      stubs: {
        // used to register custom components
        BaseButton: BaseButton,
      },
    });
  };

  beforeEach(() => {
    wrapper = factory();
  });

  afterEach(() => {
    wrapper.destroy();
  });

  test('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy();
  });

  test('match Snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('Modal to display exact title', () => {
    const title = wrapper.find('.claim-title');

    expect(title.exists()).toBe(true);
    expect(title.text()).toBe('お手元にご用意いただくもの');
  });

  test('Modal to display exact 2 step', () => {
    const items = wrapper.findAll('.claim-row');
    expect(items).toHaveLength(2);
  });

  test('Modal to display button exact text', () => {
    const component = wrapper.findComponent(BaseButton);
    expect(component.exists()).toBe(true);
  });
});
