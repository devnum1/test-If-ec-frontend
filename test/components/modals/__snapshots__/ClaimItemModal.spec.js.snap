// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClaimItemModal match Snapshot 1`] = `
"<div class=\\"claim-item-modal\\">
  <div class=\\"modal__overlay\\">
    <div class=\\"modal__container\\">
      <div class=\\"modal__body modal-claim\\">
        <div class=\\"claim-title\\">
          <div class=\\"claim-title__line\\">
            お手元にご用意いただくもの
          </div>
        </div>
        <div class=\\"claim-row\\">
          <div class=\\"claim-row__number\\">
            <div class=\\"claim-row__number__round\\">
              1
            </div>
          </div>
          <div class=\\"claim-row__text\\">
            診療明細書の写真
          </div>
          <div class=\\"claim-row__icon\\"><img src=\\"~/assets/images/icon-contract.svg\\" class=\\"claim-row__icon-image\\"></div>
        </div>
        <div class=\\"claim-row\\">
          <div class=\\"claim-row__number\\">
            <div class=\\"claim-row__number__round\\">
              2
            </div>
          </div>
          <div class=\\"claim-row__text\\">
            飼い主とペットが 一緒に写った写真
          </div>
          <div class=\\"claim-row__icon\\"><img src=\\"~/assets/images/icon-pet-owner.svg\\" class=\\"claim-row__icon-image\\"></div>
        </div> <button type=\\"button\\" class=\\"btn btn--default btn--icon-right test btn--full\\">
          <!---->
          <!----> <span class=\\"btn__text\\">保険金を請求する</span> <i class=\\"icon-fonts icon-fonts--arrow-right\\"></i>
          <!----></button>
      </div>
    </div>
  </div>
</div>"
`;
